Sources/MessagingPush/RichPush/MessagingPush+RichPush.swift

```diff
@@ -51,10 +51,9 @@ extension MessagingPushImplementation {
         logger.info("did receive notification request. Checking if message was a push sent from Customer.io...")
         logger.debug("notification request: \(request.content.userInfo)")
 
-        guard let parsedPush = CustomerIOParsedPushPayload.parse(
-            notificationContent: request.content,
-            jsonAdapter: jsonAdapter
-        ) else {
+        let push = UNNotificationWrapper(notificationRequest: request)
+
+        guard let pushCioDeliveryInfo = push.cioDelivery else {
```

Comment:
hmm `isPushSentFromCio` would have made more sense here, maybe we get this value below?

Reply:
I had the same thought when I wrote the code. I would also prefer to use `isPushSentFromCio`, but I was also trying to avoid any force optionals: 

```swift 
let push = UNNotificationWrapper(notificationRequest: request)

guard push.isPushSentFromCio else {
  return 
}

let pushCioDeliveryInfo = push.cioDelivery! <--- what I want to avoid
```

Perhaps, this would be more readable and avoid a force optional? 

```swift 
let push = UNNotificationWrapper(notificationRequest: request)

guard push.isPushSentFromCio, let pushCioDeliveryInfo = push.cioDelivery else {
  return 
}
```